validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -36,43 +36,44 @@
 
 
 class AddressNotInTree(Exception):
-    pass
+    def __init__(self, match=None):
+        super().__init__()
+        self.match = match
 
 
 class Node:
     def __init__(self, address, data=None):
         self.address = address
-        self.children = {}
+        self.children = set()
         self.data = data
 
     def __repr__(self):
-        return repr({
-            'address': self.address,
-            'children': self.children,
-            'data': self.data,
-        })
+        return repr((
```

Comment:
Since you're modifying this content, perhaps this should be changed to actually be the correct usage of `__repr__` vs what this looks like, which more of a debugging string.

See: https://docs.python.org/3.5/library/functions.html#repr

Reply:
Why do we even both implementing this? I have always considered __repr__ and __str__ to just be for debugging purposes. If you don't implement __repr__ or __str__, a default implementation is provided which "is a string enclosed in angle brackets that contains the name of the type of the object ..." Do you mean this should be reimplemented so that it is "an attempt to return a string that would yield an object with the same value when passed to eval()"? That seems kind of absurd since you should never be passing this to eval() unless you are copy-pasting some logging message directly into an interactive python interpreter.

Reply:
I don't think `__repr__` here was ever anything other than a debug string.

Reply:
Change it to `__str__`, then